File: frontend/lib/features/tracking/models/cv_nutrition_info.dart
class CvNutritionInfo {
  CvNutritionInfo({
    required this.tongCalories,
    required this.proteinG,
    required this.carbsG,
    required this.fatG,
    required this.fiberG,
  });

  final double tongCalories;
  final double proteinG;
  final double carbsG;
  final double fatG;
  final double fiberG;

  factory CvNutritionInfo.fromJson(Map<String, dynamic> json) {
    return CvNutritionInfo(
      tongCalories: _number(json['tong_calories'] ?? json['tongCalories']),
      proteinG: _number(json['protein_g'] ?? json['proteinG']),
      carbsG: _number(json['carbs_g'] ?? json['carbsG']),
      fatG: _number(json['fat_g'] ?? json['fatG']),
      fiberG: _number(json['fiber_g'] ?? json['fiberG']),
    );
  }
}

double _number(dynamic value) => value is num ? value.toDouble() : double.tryParse('$value') ?? 0;

File: backend/MenuGreen.BusinessLogicLayer/DTOs/Responses/CvNutritionInfo.cs
using System.Text.Json.Serialization;

namespace MenuGreen.BusinessLogicLayer.DTOs.Responses
{
    public class CvNutritionInfo
    {
        [JsonPropertyName("tong_calories")]
        public double TongCalories { get; set; }

        [JsonPropertyName("protein_g")]
        public double ProteinG { get; set; }

        [JsonPropertyName("carbs_g")]
        public double CarbsG { get; set; }

        [JsonPropertyName("fat_g")]
        public double FatG { get; set; }

        [JsonPropertyName("fiber_g")]
        public double FiberG { get; set; }
    }
}
